src/components/results/result.js

```diff
@@ -11,6 +11,27 @@ class Result extends React.Component {
     return value.join(", ");
   }
 
+
+  // Highlight the search words in the result text.
+  HighlightWords(field, text){
+    if (field.type === 'text' && field.value !== undefined){
+      // replace "&", "|", "(", ")", "*", "'", """ and duplicated whitespace with only one whitespace
+      let values = field.value.replaceAll(/\&|\*|\(|\)|\||\"|\'/g,"").replaceAll(/\s+/g," ").split(" ");
+
+      // split original text by search words insensitively by regex pattern
+      // g is for regex global and i is for insensitive.
+      values.map((value) => {
+        let regex = new RegExp(value, "gi");
+        text = text.split(regex).join("<span style='background-color:yellow;'>" + value + "</span>");
+      })
+
+    }
+
+    // How to render HTML string in React: https://medium.com/@uigalaxy7/how-to-render-html-in-react-7f3c73f5cafc
+    // dangerouslySetInnerHTML is to convert html string into react html object 
+    return <span dangerouslySetInnerHTML={{__html: text}} />;
```

Comment:
In perusing the docs for this, it indicates that this vulnerable to XSS attacks.  I'm not an expert here, but should we see if there's an accepted safe approach to using this method?

Reply:
My current thought is to modify the content in other files rather than the library after the solr-faceted-search-react renders the result. I'll try if it works.

Reply:
That's probably a great idea!